source/common/http/async_client_impl.cc

```diff
@@ -39,6 +39,7 @@ AsyncClient::Request* AsyncClientImpl::send(MessagePtr&& request, AsyncClient::C
                                             const Optional<std::chrono::milliseconds>& timeout) {
   AsyncRequestImpl* async_request =
       new AsyncRequestImpl(std::move(request), *this, callbacks, timeout);
+  async_request->initialize();
```

Comment:
AsyncStreamImpl doesn't currently do any real work in its ctor, though. I could add an `initialize()` method but it would only be for consistency's sake.